src/controllers/searchResult/base.js

```diff
@@ -97,6 +95,46 @@ class SearchResultBase {
     throw new Error('Unable to retrieve next results from search: missing scrollId, from/sort, or from/size params');
   }
 
+  /**
+   * Automatically fetch each page of results and execute the provided action
+   * on each hit.
+   *
+   * If the action return a promise, this function will wait for all promise to
+   * be resolved.
+   *
+   * @param {Function} action - Action to execute for each hit
+   * @returns {Promise}
+   */
+  async forEachHit(action, firstCall = true) {
+    let results;
+
+    if (firstCall) {
+      results = new this.constructor(this._kuzzle, this._request, this._options, this._response);
```

Comment:
why do we need to instantiate a new `SearchResult` object here ?

Reply:
The `next` method does not return a new `SearchResult` but instead it modify the properties.  
I instantiate a new `SearchResult` to avoid mutating the original one when calling `forEachHit`

Reply:
I agree, the `next` method should return a new `SearchResult` instance, as it is done in other SDK, instead of mutating the current object.